api/src/test/java/org/openmrs/notification/db/hibernate/HibernateAlertDAOTest.java

```diff
@@ -32,11 +46,49 @@ public void runBeforeEachTest() {
 	 * @throws Exception
 	 */
 	@Test
-	public void shouldGetAlerts() {
-		
-		AlertService as = Context.getAlertService();
-		//System.out.println(as.getAllAlerts());
+	public void getAlert_shouldReturnSavedAlert() throws Exception {
```

Comment:
Is this a test or its just a class, If its a test ,could you kindly make it a full test by adding @test annotation

Reply:
hi @sherrif10   thanks for the review,the above method already has a @test annotation cc @ibacher

Reply:
So a quick thought on this. We want tests to only test a single "unit" of work. So, for example, the code here is basically identical to the code to test `saveAlert()` which tells me that something has gone wrong. I would model these tests on the tests that we have for [`HibernateConceptDAOTest`](https://github.com/openmrs/openmrs-core/blob/52598a4f44882107a14aa680175c2e094a9fa761/api/src/test/java/org/openmrs/api/db/hibernate/HibernateConceptDAOTest.java#L1).

In particular, this uses DBUnit to put the database in a known state and adds the `@Autowired` annotation to the `dao` being tested.

Then, the test for the various `get` methods becomes *just* a matter of getting known values from the database (because they are configured using a DBUnit XML document) and so if we break the save functionality, it doesn't also break this test.

Also on naming conventions, the key part about the `getAlert()` method is that it gets the alert by the id field so I would name this something like:

```java
public void getAlert_shouldGetAlertById()
```

Similarly, `saveAlert()` probably needs at least two tests:

```java
public void saveAlert_shouldSaveAlertToDb()

public void saveAlert_shouldRaiseExceptionWhenAlertInvalid()
```